src/main/java/org/deepsymmetry/beatlink/AnnouncementSocketConnection.java

```diff
@@ -0,0 +1,294 @@
+                                                packet.getLength() + ".");
+                                    } else {
+                                        if (packet.getLength() > 54) {
+                                            logger.warn("Processing too-long " + kind.name + " packet; expected 54 bytes, but got " +
+                                                    packet.getLength() + ".");
+                                        }
+                                        DeviceAnnouncement announcement = new DeviceAnnouncement(packet);
+
+                                        deliverDeviceAnnouncement(announcement);
+                                    }
+                                } else if (kind == Util.PacketType.DEVICE_HELLO) {
+                                    logger.debug("Received device hello packet.");
+                                } else if (kind != null) {
+                                    // TODO, probably should be a set of listeners to not couple VirtualCdj to this class
+                                    VirtualCdj.getInstance().handleSpecialAnnouncementPacket(kind, packet);
+                                }
+                            }
+                        } catch (Throwable t) {
+                            logger.warn("Problem processing DeviceAnnouncement packet", t);
+                        }
+                    }
+                }
+            }, "beat-link DeviceFinder receiver");
+            receiver.setDaemon(true);
+            receiver.start();
+        }
+    }
+
+    public void send(DatagramPacket packet) throws IOException {
+        announcementSocket.get().send(packet);
+    }
+
+    @Override
+    public SocketSender getSocketSender() {
+        return this;
+    }
+
+
+    /**
+     * Keeps track of the registered device announcement listeners.
+     */
+    private final Set<AnnouncementListener> deviceListeners =
+            Collections.newSetFromMap(new ConcurrentHashMap<AnnouncementListener, Boolean>());
+
+    /**
+     * Adds the specified device announcement listener to receive device announcements when DJ Link devices
+     * are found on or leave the network. If {@code listener} is {@code null} or already present in the list
+     * of registered listeners, no exception is thrown and no action is performed.
+     *
+     * <p>Device announcements are delivered to listeners on the
+     * <a href="https://docs.oracle.com/javase/tutorial/uiswing/concurrency/dispatch.html">Event Dispatch thread</a>,
+     * so it is fine to interact with user interface objects in listener methods. Any code in the listener method
+     * must finish quickly, or unhandled events will back up and the user interface will be come unresponsive.</p>
+     *
+     * @param listener the device announcement listener to add
+     */
+    public void addDeviceAnnouncementListener(AnnouncementListener listener) {
```

Comment:
We should rename these methods to match the new interface name, either `addAnnouncementListener`, or if you agree with my `AnnouncementPacketListener` name for the new low-level interface, `addAnnouncementPacketListener`, and the same for the `delete…` method.

Reply:
Good call. Done.